### PR TITLE
I ported Scratch to OpenFL

### PR DESCRIPTION
**Pull request by my2iu**
I made a basic port of the Scratch code to Haxe/OpenFL. I commented out a lot of code to make the port easier though. I encountered a couple of minor bugs in OpenFL during the port. I don't think a merge of the code is possible, but I just wanted to let you know.

https://github.com/my2iu/scratch-flash/tree/openfl-port

### CODE REVIEW

**Comment by jgranick** [comment]
This is awesome! I think we should get it running in Flash as well, if we can. I've sent you an invite to this organization and team if you want, to continue to contribute directly. Seeing this run in HTML5 so far is SO AWESOME. Thank you!

**Comment by my2iu** [comment]
Thanks for the invite. As far as I can tell, I think the Scratch people aren't actually interested in people porting Scratch to html5 or other systems in the end, so I'm probably going to hold off from doing further development on the port. I don't want to put a lot of work into porting it, just to have all that work go to waste. As such, I'm going to decline the invite for now. Thanks though!

**Comment by jgranick** [comment]
Is there a place where you had this discussion?

The Raspberry Pi includes Scratch 1 by default, but not Scratch 2 because it does not run Flash Player. As it is possible to run OpenFL on a Raspberry Pi, I think that's one example (beyond HTML5) of why a port would be valuable :smile:

**Comment by my2iu** [comment]
Oh, it's more that I recently realized that the Scratch team chose to use an open source license that prevents them from using any community changes on the Scratch website. Scratch is released under the GPL, but the version of Scratch used on their website contains proprietary components, so they can't use any changes made by the community. If they wanted to incorporate any changes from others, they would have to reprogram everything themselves. Since they appear to have purposely designed things this way, I don't think they're interested in ports. I don't think it's worth spending 2-6 more weeks of full-time effort to complete the port only to have them redo all the effort again if they actually become interested in an HTML5 port of Scratch.

Basically, without know what the intentions of the main Scratch team are, I don't know whether it's worth spending all the time to finish the port. If they have their own plans for an HTML5 port, then there's no point duplicating their effort. If this is the only way we'll ever get an HTML5 port, then it's probably worth doing. I sent them a note about the port to their GitHub, but I haven't heard anything back.
